src/payment/strategies/braintree/braintree-payment-processor.ts

```diff
@@ -38,21 +46,19 @@ export default class BraintreePaymentProcessor {
             }));
     }
 
-    paypal(amount: number, storeLanguage: string, currency: string, offerCredit: boolean): Promise<NonceInstrument> {
+    paypal({ shouldSaveInstrument, ...config }: PaypalConfig): Promise<BraintreeTokenizePayload> {
         return this._braintreeSDKCreator.getPaypal()
             .then(paypal => {
                 this._overlay.show({
                     onClick: () => paypal.focusWindow(),
                 });
 
                 return paypal.tokenize({
-                    amount,
-                    currency,
                     enableShippingAddress: true,
-                    flow: 'checkout',
-                    locale: storeLanguage,
-                    offerCredit,
+                    flow: shouldSaveInstrument ? 'vault' : 'checkout',
+                    offerCredit: false,
```

Comment:
Do you need `offerCredit: false` when `config` already provides that value?

Reply:
sorted